python/understack-workflows/understack_workflows/nautobot_device.py

```diff
@@ -139,26 +138,26 @@ def switches_for(nautobot, chassis_info: ChassisInfo) -> dict:
 def nautobot_switches(nautobot, mac_addresses: set[str]) -> dict[str, dict]:
     """Get switches by MAC address.
 
+    Assumes switch MAC addresses are present in Nautobot in a custom field on
+    Device called chassis_mac_address.
+
     Assumes that MAC addresses in Nautobot are normalized to upcase
     AA:BB:CC:DD:EE:FF form.
 
-    We store the MAC address in the nautobot device "asset tag" field because
-    there was nowhere else to put it.
-
     returns a dict[mac_address] -> dict switch information indexed by mac
     """
-    formatted_list = json.dumps(list(mac_addresses))
+    pattern = "|".join(mac_addresses)
 
     query = (
         """{
-        devices(asset_tag: %s){
+        devices(cf_chassis_mac_address__re: "(%s)"){
             id name
-            mac: asset_tag
+            mac: cf_chassis_mac_address
             location { id name }
             rack { id name }
         }
     }"""
-        % formatted_list
+        % pattern
```

Comment:
string interpolation creates the GraphQL injection vulnerability. For example, one could say

```python
mac_addresses='{"malicious_input")} { devices { id name } } #"}'
```

Consider replacing this with query variables (untested):
```python
query = """
query($pattern: String!) {
    devices(cf_chassis_mac_address__re: $pattern) {
        id name
        mac: cf_chassis_mac_address
        location { id name }
        rack { id name }
    }
}
"""

variables = {"pattern": "|".join(mac_addresses)}
result = nautobot.graphql.query(query, variables=variables)
```